HACKING.md

```diff
@@ -0,0 +1,409 @@
+
+    pyenv virtualenv 3.7.0 salt
+
+Then activate it:
+
+    pyenv activate salt
+
+Sweet! Now you're ready to clone salt so you can start hacking away!
+
+### Cloning Salt
+
+Clones are so shallow. Well, this one is anyway:
+
+    git clone --depth=1 --origin salt https://github.com/saltstack/salt.git
+
+This creates a shallow clone of salt, which should be fast. Most of the time
+that's all you'll need, and you can start building out other commits as you go.
+If you *really* want all 108,300+ commits you can just run `git fetch
+--unshallow`. Then go make a sandwich because it's gonna be a while.
+
+### Fork Salt
+
+You're also going to want to head over to GitHub and create your own [fork of
+salt](https://github.com/saltstack/salt/fork). Once you've got that setup you
+can add it as a remote:
+
+    git remote add yourname <YOUR SALT REMOTE>
+
+If you use your name to refer to your fork, and `salt` to refer to the official
+Salt repo you'll never get `upstream` or `origin` confused.
+
+### `pre-commit` and `nox` Setup
+
+Here at Salt we use [pre-commit](https://pre-commit.com/) and
+[nox](https://nox.thea.codes/en/stable/) to make it easier for contributors to
+get quick feedback. Nox enables us to run multiple different test
+configurations, as well as other common tasks. You can think of it as Make with
+superpowers. Pre-commit does what it sounds like - it configures some Git
+pre-commit hooks to run `black` for formatting, `isort` for keeping our imports
+sorted, and `pylint` to catch issues like unused imports, among others. You can
+easily install them in your virtualenv with:
+
+    python -m pip install pre-commit nox
+    pre-commit install
+
+Now before each commit, it will ensure that your code at least *looks* right
+before you open a pull request. And with that step, it's time to start hacking
+on Salt!
+
+
+## Selecting an Issue
+
+If you don't already have an issue in mind, you can search for [help
+wanted](https://github.com/saltstack/salt/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22)
+issues. If you also search for [good first
+issue](https://github.com/saltstack/salt/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22+label%3A%22good+first+issue%22)
+then you should be able to find some issues that are good for getting started
+contributing to Salt. [Documentation
+issues](https://github.com/saltstack/salt/issues?q=is%3Aissue+is%3Aopen+label%3Adocumentation+)
+are also good starter issues. When you find an issue that catches your eye (or
+one of your own), it's a good idea to comment on the issue and mention that
+you're working on it. Good communication is key to collaboration - so if you
+don't have time to complete work on the issue, just leaving some information
+about when you expect to pick things up again is a great idea!
+
+## Hacking Away
+
+### Salt, Tests, Documentation, and You
+
+To merge code contributions, Salt requires:
+
+- documentation
+- meaningful passing tests
+- correct code
+
```

Comment:
**Nit-pick (blocking):** I don't think that's all that is required to merge. This is what is required in a pull request, yes, but NOT to merge.

You can correct me if I'm wrong, but to merge you need:

- At least 1 approval from a Core developer (and then we need to indicate here that the bot automatically assigns a Core developer to review the PR, but I think we also need an about page that says who the Core developers are)
- All tests must pass
- If tests don't pass, they need 3 approvals from Core developers

And we need something that sets expectations for how the team will review them. The truth is that it will likely take the team a long time to get to their PR. We need to find some way to communicate that. I also don't know how they can go about getting their PR escalated. My hunch is that if they are a trusted member of the community and an active contributor who has shown a long-term commitment to the project (by participating in working groups, answering questions on Slack, joining Open Hours, etc.) that helps tremendously.

Reply:
Also, it's unclear to me what to do after you've met these requirements. Who merges and how do you tell them that you're ready to merge because you've met the requirements? Do they need to take any extra action? Do they need to constantly rebase their pull request while they wait? What if rebasing causes a new test to fail?